### PR TITLE
Use shell icon if file is executable and has no extension - also new icons

### PR DESCRIPTION
**Pull request by Babkock**
Hello, I hope this pull request finds you well. I really like the tool and I had an idea for how to give executable binaries a special icon. I also added a lot of icons, file names, and extensions, that the developers of this project don't seem to have thought of. I have segregated my contributions into 3 commits, the first one has the actual code change, the second one has only additions to the file name and extension lists, and the third has subjective changes and more additions.

Some of the changes I've made:

* Changed the folder icon
* PHP uses elephant logo
* Special file names like `Cargo.toml`, `package.json`, `webpack.config.js`, `Makefile`, and `composer.json` now have icons that match their respective purposes
* Video files use film strip instead of video camera
* Markdown files use down arrow instead of "M" + down arrow
* Better logo for R source files
* Added lesser-known extensions, such as `.s`, `.o`, `.so`, `.a`, `.1`
* Added a package icon for Linux software packages: `.deb`, `.pkg`, `.rpm`, `.xbps`, `.img`, `.iso`
* Added icons for lesser-known media formats

I hope everyone is okay with these changes; if not, I can revert them. I thought I would include these because I personally like the way certain icons look better than others, but I understand this is subjective. Do you like these?

### CODE REVIEW

**Comment by Babkock** [comment]
Hi, I have added another commit that should fix the CI tests.

File: src/icon.rs [hunk 1]
```diff
@@ -30,7 +30,7 @@ impl Icons {
                     Self::get_default_icons_by_name(),
                     Self::get_default_icons_by_extension(),
                     "\u{f016}", // 
-                    "\u{f115}", // 
```
Comment (zwpaper) [inline src/icon.rs:33]: do not change the default icon, if want to, we can raise another issue and discuss it together.

File: src/icon.rs [hunk 2]
```diff
@@ -105,21 +108,35 @@ impl Icons {
...
+        m.insert("cargo.toml", "\u{e7a8}"); // ""
+        m.insert("cargo.lock", "\u{e7a8}"); // ""
+        m.insert("changelog", "\u{e609}"); // ""
+        m.insert("composer.json", "\u{e608}"); // ""
         m.insert("config", "\u{e5fc}"); // ""
+        m.insert("config.mk", "\u{e615}"); // ""
         m.insert("docker-compose.yml", "\u{f308}"); // ""
         m.insert("dockerfile", "\u{f308}"); // ""
         m.insert("ds_store", "\u{f179}"); // ""
+        m.insert(".emacs.d", "\u{e615}"); // ""
```
Comment (zwpaper) [inline src/icon.rs:138]: how about the GNU icon `e779` for Emacs

File: src/icon.rs [hunk 3]
```diff
@@ -212,7 +256,7 @@ impl Icons {
         m.insert("gform", "\u{f298}"); // ""
         m.insert("gif", "\u{f1c5}"); // ""
         m.insert("git", "\u{f1d3}"); // ""
-        m.insert("go", "\u{e626}"); // ""
+        m.insert("go", "\u{e627}"); // ""
```
Comment (zwpaper) [inline src/icon.rs:259]: I remember we have a issue before discussing this `Go` icon seems a little bit creepy for the eyes in some background, maybe we could keep the origin one

File: src/icon.rs [hunk 4]
```diff
@@ -253,26 +302,31 @@ impl Icons {
         m.insert("m3u", "\u{f910}"); // "蘿"
         m.insert("m3u8", "\u{f910}"); // "蘿"
         m.insert("m4a", "\u{f001}"); // ""
+        m.insert("m4v", "\u{f008}"); // ""
         m.insert("magnet", "\u{f076}"); // ""
-        m.insert("markdown", "\u{f48a}"); // ""
-        m.insert("md", "\u{f48a}"); // ""
+        m.insert("markdown", "\u{e609}"); // ""
```
Comment (zwpaper) [inline src/icon.rs:307]: the origin Markdown icon seems more popular, maybe we should keep the origin one

**Review by zwpaper — CHANGES_REQUESTED** [review]
Thanks for adding so many icons! but for the existed part, we could have some discussion, comments added

File: src/icon.rs (hunk 3 above)
Comment (meain) [reply src/icon.rs:259]: ref: https://github.com/Peltoche/lsd/pull/333
PS: I don't have any strong opinions on this anymore

File: src/icon.rs (hunk 1 above)
Comment (Babkock) [reply src/icon.rs]: Okay sounds good. I changed it back, and the tests, too.

File: src/icon.rs (hunk 2 above)
Comment (Babkock) [reply src/icon.rs]: Sounds good

File: src/icon.rs (hunk 4 above)
Comment (Babkock) [reply src/icon.rs]: Sounds good, it has been reverted. I personally like just the arrow

**Comment by Babkock** [comment]
How do I fix the failing checks? Running `cargo test` on my computer shows that all tests pass... is there something I have to do? I don't have Windows, but Void Linux. Do I make more commits?

**Comment by zwpaper** [comment]
you can check the detail by clicking the `Details` link on the right side of each failed test.

it is a style check, you can check it by `cargo fmt && cargo clippy -D warnings`

**Comment by meain** [comment]
Merging it with some changes. Thanks for the PR :D

**Pull request merged**